lib/ansible/modules/database/postgresql/postgresql_facts.py

```diff
@@ -0,0 +1,921 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Andrew Klychkov (@Andersson007) <aaklychkov@mail.ru>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: postgresql_facts
+short_description: Gather facts about remote PostgreSQL servers
+description:
+- Gathers facts about remote PostgreSQL servers.
+version_added: "2.8"
+options:
+  filter:
+    description:
+    - Limit collected facts by comma separated string or YAML list.
+    - Allowable values are C(version),
+      C(databases), C(settings), C(tablespaces), C(languages), C(roles), C(namespaces),
+      C(replications), C(repl_slots), C(extensions).
+    - By default, collects all subsets.
+    - You can use shell-style (fnmatch) wildcard to pass groups of values (see Examples).
+    - You can use '!' before value (for example, C(!settings)) to exclude it from facts.
+    - If you pass including and excluding values to the filter, for example, I(filter=!settings,ver),
+      the excluding values will be ignored.
+    type: list
+  db:
+    description:
+    - Name of database to connect.
+    type: str
+  port:
+    description:
+    - Database port to connect.
+    type: int
+    default: 5432
+  login_user:
+    description:
+    - User (role) used to authenticate with PostgreSQL.
+    type: str
+    default: postgres
+  login_password:
+    description:
+    - Password used to authenticate with PostgreSQL.
+    type: str
+  login_host:
+    description:
+    - Host running PostgreSQL.
+    type: str
+  login_unix_socket:
```

Comment:
I'd prefer to remain it for compatibility

Reply:
Compatibility with what?

Reply:
Maybe somebody use this option. It is presented in other PG modules.
If you absolutely sure that it's useless, OK, I'll remove it.
Are you sure? ;)

Reply:
Sorry, I had no idea old PG modules also had that option, because in PG clients `hosts` parameter was always set to the path of the unix socket. I guess it's better to leave it, and maybe in the next version we can make it an alias to host or something similar for all modules.